### PR TITLE
Feature/rb 94

### PR DESCRIPTION
**Pull request by Saverio976**
* **Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [x] Norm of the code has been respected

* **In what state is your pull request?**
- [x] Ready to merge / Waiting a reviwer to see my work.
- [ ] Work In Progress (WIP) / My work is not finish but i want daily reviews. (Draft)
- [ ] CI Review / I want to see what the CI thinks of my work.
- [ ] Need feedback / Just to have feedback on what i produced. (May not be merged)

* **What kind of change does this PR introduce?** (You can choose multiple)
- [ ] Bug fix
- [x] Feature request
- [ ] New / Updated documentation
- [ ] Testing CI ( Make the pull request in draft mode)

* **What is the current behavior?** (link an issue based on the kind of change this pr introduce)

* **What is the new behavior (if this is a feature change)?**

* **Other information**:

### Summary by CodeRabbit

- New Feature: Added a new message type to handle relative position updates, improving real-time interaction in the game.
- Refactor: Simplified the process of sending data by using a more specific `Packet` type instead of `std::any` and `boost::asio::ip::udp::endpoint` parameters. This change enhances the code's readability and maintainability.
- New Feature: Introduced a player ID system to track individual players, enhancing the game's multiplayer capabilities.
- Refactor: Updated various functions to use the new `Packet` constructor, streamlining the process of creating and sending packets.
- New Feature: Added a new enemy type, `CLASSIC_ENEMY`, expanding the game's variety.
  
Please note that these changes are primarily backend improvements and may not result in noticeable changes for end-users. However, they lay the groundwork for future enhancements and features.

### CODE REVIEW

File: src/Nitwork/NitworkServer.cpp [hunk 1]
```diff
@@ -130,6 +130,38 @@ namespace Nitwork {
         addStarWaveMessage(endpoint, Types::Enemy::getEnemyNb());
         Systems::SystemManagersDirector::getInstance().getSystemManager(0).addSystem(Systems::initWave);
     }
+
+    void NitworkServer::handleRelativePositionMsg(
+        const std::any &msg,
+        boost::asio::ip::udp::endpoint &endpoint /* unused */)
```
Comment (romainpanno) [inline src/Nitwork/NitworkServer.cpp]: not unused

**Review by romainpanno — CHANGES_REQUESTED** [review]

File: src/ECS/MessageTypes.h [hunk 2]
```diff
@@ -9,6 +9,8 @@
         #define PACK( __Declaration__ ) __pragma( pack(push, 1) ) __Declaration__ __pragma( pack(pop))
     #endif
 
+typedef unsigned int n_id_t;
```
Comment (guillaumeAbel) [inline src/ECS/MessageTypes.h:12]: exist in Nitwork.h

**Review by guillaumeAbel — APPROVED** [review]

File: src/ECS/MessageTypes.h (hunk 2 above)
Comment (Saverio976) [reply src/ECS/MessageTypes.h:12]: nop, j'lai suprr

Comment (Saverio976) [reply src/ECS/MessageTypes.h:12]: par ce que nitwork.h inclus MessageTypes.h

File: src/Nitwork/NitworkServer.cpp [hunk 3]
```diff
@@ -142,11 +170,10 @@ namespace Nitwork {
             .msg    = {.magick = MAGICK_INIT, .playerId = playerId}
         };
         Packet packet(
-            packetMsgPlayerInit.header.id,
             packetMsgPlayerInit.action.magick,
-            std::make_any<struct packetMsgPlayerInit_s>(packetMsgPlayerInit),
-            getEndpointSender());
-        addPacketToSend(endpoint, packet);
+            std::make_any<struct packetMsgPlayerInit_s>(packetMsgPlayerInit));
```
Comment (romainpanno) [inline src/Nitwork/NitworkServer.cpp]:
```suggestion
            std::make_any<struct packetMsgPlayerInit_s>(packetMsgPlayerInit),
            endpoint);
```

**Review by romainpanno — CHANGES_REQUESTED** [review]

**Review by romainpanno — APPROVED** [review]

**Pull request merged**